crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1862,12 +1866,11 @@ where
                     continue
                 }
 
-                // Process one incoming message from the CL. We don't drain the messages right away,
-                // because we want to sneak a polling of running hook in between them.
+                // Process one incoming message from the CL.
                 //
                 // These messages can affect the state of the SyncController and they're also time
                 // sensitive, hence they are polled first.
-                if let Poll::Ready(Some(msg)) = this.engine_message_stream.poll_next_unpin(cx) {
+                while let Some(msg) = this.queued_engine_messages.pop_front() {
```

Comment:
the engine message processing here only sets a single action to be performed in the blockchain tree. you cannot do it in a while loop because it overwrites the last set action

Reply:
didn't work before that commit either :/

Reply:
https://github.com/paradigmxyz/reth/pull/8238/checks?sha=aa306d7824bbeedf6b680b8014a1088031266ac1